### PR TITLE
[CALCITE-5863] Calcite rejects valid query with multiple ORDER BY columns and constant RANGE bounds in window functions

### PR DESCRIPTION
**Pull request by itiels**
Allow multiple ORDER BY columns with RANGE bounds in window functions, if both of the bounds are constant bounds (CURRENT ROW, UNBOUNDED PRECEDING, UNBOUNDED FOLLOWING)

### CODE REVIEW

**Comment by LakeShen** [comment]
Hi,could you make the title CALCITE-5863 to [CALCITE-5863]

File: core/src/main/java/org/apache/calcite/sql/SqlWindow.java
Comment (LakeShen) [inline core/src/main/java/org/apache/calcite/sql/SqlWindow.java:613]: Can you add a sql case to the misc.iq file to test the results？

Comment (itiels) [reply core/src/main/java/org/apache/calcite/sql/SqlWindow.java:614]: Added test case

File: core/src/main/java/org/apache/calcite/sql/SqlWindow.java
Comment (mihaibudiu) [inline core/src/main/java/org/apache/calcite/sql/SqlWindow.java:667]: This does not seem to check for what the compiler calls "literal", which is essentially a constant value.
You are checking for specific kinds of bounds. So perhaps this function is not named properly.

Comment (itiels) [reply core/src/main/java/org/apache/calcite/sql/SqlWindow.java]: Naming is hard.. My thought was literal in the sense of `SqlLiteral`, but I can see why it's confusing. 
Changed it to `onlySymbolBounds`.

**Review by mihaibudiu — APPROVED** [review]

**Comment by JiajunBernoulli** [comment]
Nice work, can we add some tests in RelOptRulesTest?
> Use some window rule like `CoreRules.WINDOW_REDUCE_EXPRESSIONS`.

**Comment by itiels** [comment]
@JiajunBernoulli I'm assuming you've wanted a test to see that the `WINDOW_REDUCE_EXPRESSIONS` rule can handle the queries that were rejected by the validator before, as I haven't changed any rules code, So I've added such a test. please let me know if that's not what you've meant.

**Comment by libenchao** [comment]
This has been reviewed by 3 committers, and with 1 approval. Could we merge it within 1.36.0?

**Comment by LakeShen** [comment]
Hi @itiels ,please keep the same title with jira's title.

I find the jira's title is `Calcite rejects valid query with multiple ORDER BY columns and constant RANGE bounds in window functions`

**Comment by LakeShen** [comment]
At the same time,the commits' information should be same with jira's title.

**Comment by JiajunBernoulli** [comment]
I will approve if commit message  same with JIRA summary.

**Comment by itiels** [comment]
@JiajunBernoulli @LakeShen should I squash the commits and rewrite the commit message?

**Comment by itiels** [comment]
@LakeShen done. Thank you!

**Review by LakeShen — APPROVED** [review]

**Pull request merged**
